File: apps/mobile/modules/pose-detector/android/src/main/java/expo/modules/posedetector/PoseFrameProcessorPlugin.kt
package expo.modules.posedetector

import com.mrousavy.camera.frameprocessors.Frame
import com.mrousavy.camera.frameprocessors.FrameProcessorPlugin
import expo.modules.kotlin.AppContext

class PoseFrameProcessorPlugin(private val appContext: AppContext) : FrameProcessorPlugin() {
  override fun callback(frame: Frame, params: Map<String, Any>?): Any? {
    val context = appContext.reactContext ?: return null
    return PoseLandmarkerHelper.detect(context, frame)
  }
}

File: apps/mobile/modules/pose-detector/ios/PoseFrameProcessorPlugin.swift
import UIKit
import VisionCamera

@objc(PoseFrameProcessorPlugin)
public class PoseFrameProcessorPlugin: FrameProcessorPlugin {
  public override func callback(_ frame: Frame, withArguments _: [AnyHashable: Any]?) -> Any? {
    return PoseLandmarkerHelper.shared.detect(sampleBuffer: frame.buffer)
  }
}
